lms/djangoapps/certificates/badge_handler.py

```diff
@@ -0,0 +1,151 @@
+
+    def badge_url(self, mode):
+        """
+        Get the URL for a course's badge in a given mode.
+        """
+        return "{}/{}".format(self.badge_create_url, self.course_slug(mode))
+
+    def assertion_url(self, mode):
+        """
+        URL for generating a new assertion.
+        """
+        return "{}/assertions".format(self.badge_url(mode))
+
+    def course_slug(self, mode):
+        """
+        Slug ought to be deterministic and limited in size so it's not too big for Badgr.
+        """
+        return hashlib.sha256(u"{}{}".format(unicode(self.course_key), unicode(mode))).hexdigest()
+
+    def log_if_raised(self, response):
+        """
+        Log server response if there was an error.
+        """
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            LOGGER.error(
+                u"Encountered an error when contacting the Badgr-Server. Response status was %s.\n%s",
+                response.status_code, response.body
+            )
+            raise
+
+    def get_headers(self):
+        """
+        Headers to send along with the request-- used for authentication.
+        """
+        return {'Authorization': 'Token {}'.format(settings.BADGR_API_TOKEN)}
+
+    def ensure_badge_created(self, mode):
+        """
+        Verify a badge has been created for this mode of the course, and, if not, create it
+        """
+        if self.course_slug(mode) in BadgeHandler.badges:
+            return
+        response = requests.get(self.badge_url(mode), headers=self.get_headers())
+        if response.status_code != 200:
+            self.create_badge(mode)
+        BadgeHandler.badges[self.course_slug(mode)] = True
+
+    def create_badge(self, mode):
+        """
+        Create the badge spec for a course's mode.
+        """
+        course = modulestore().get_course(self.course_key)
+        if mode not in settings.BADGR_IMAGE_SOURCES:
+            raise KeyError("No image specified for course mode {} in BADGR_IMAGE_SOURCES!".format(mode))
+        image_name = settings.BADGR_IMAGE_SOURCES[mode]
+        image = open(image_name, 'rb')
+        # We don't want to bother validating the file any further than making sure we can detect its MIME type,
+        # for HTTP. The Badgr-Server should tell us if there's anything in particular wrong with it.
+        content_type, __ = mimetypes.guess_type(image_name)
+        if not content_type:
+            raise ValueError(
+                "Could not determine content-type of image! Make sure it is a properly named .png or .svg file."
+            )
+        files = {'image': (image_name, image, content_type)}
+        try:
+            domain = unicode(Site.objects.get_current().domain)
+        except Site.DoesNotExist:
+            domain = u'example.com'
+        about_path = reverse('about_course', kwargs={'course_id': unicode(self.course_key)})
+        data = {
+            'name': course.display_name,
```

Comment:
@Kelketek Is there any translation in certificates? This looks like an issue that would appear in other places when generating certificates. Otherwise we could simply use the platform default language for now - since this would be at the course level, unless a course had a way to specify its language, there wouldn't be a way to tell which language to use here, even outside of the certificate generation?